base/permuteddimsarray.jl

```diff
@@ -148,9 +148,28 @@ end
 
 Permute the dimensions of the matrix `m`, by flipping the elements across the diagonal of
 the matrix. Differs from `LinearAlgebra`'s [`transpose`](@ref) in that the
-operation is not recursive.
+operation is not recursive, which is especially useful for arrays of non-numeric values
+(where the recursive `transpose` would throw an error) and/or 2d arrays that do not represent
+linear operators.
 
 # Examples
+Unlike `transpose`, `permutedims` can be used to swap rows and columns of 2d arrays of
```

Comment:
Should this be added to the docstring for `permutedims(A::AbstractArary, perm)` instead, as that's the topmost docstring when one looks up help for `permutedims`? This might improve the visibility if one doesn't read it all the way through

Reply:
Maybe it would be clearer to simply merge the docs in the two cases.  i.e. have a docstring like
```
    permutedims(A::AbstractArray, perm)
    permutedims(A::AbstractMatrix) --> permutedims(A::AbstractMatrix, (2,1))

Permute the dimensions of array `A`. `perm` is a vector or a tuple of length `ndims(A)`
specifying the permutation.   If `A` is a 2d array (an [`AbstractMatrix`](@ref)), then 
the `perm` argument defaults to swapping the rows and columns of the matrix.

Unlike `transpose`, …
```

Reply:
Or maybe we should just add a "see also"?

Reply:
Merging makes sense to me, as these methods are essentially performing the same operation